sdk/core/azure-core/src/main/java/com/azure/core/models/CloudEventDataFormat.java

```diff
@@ -15,6 +15,15 @@
  * @see CloudEvent#CloudEvent(String, String, BinaryData, com.azure.core.models.CloudEventDataFormat, String)
  */
 public final class CloudEventDataFormat extends ExpandableStringEnum<CloudEventDataFormat> {
+    /**
+     * Creates a new instance of {@link CloudEventDataFormat} without a {@link #toString()} value.
+     * <p>
+     * This constructor shouldn't be called as it will produce a {@link CloudEventDataFormat} which doesn't
+     * have a String enum value.
+     */
+    public CloudEventDataFormat() {
```

Comment:
Yeah, both are breaking - one is compile-time and the other is a runtime breaking change. Making the ctor private will mean revapi flagging it, requiring a major version update and API view approval. 

Ideally, we should have private ctors (which should be our guidance from now on for all new expandable enums) but not sure if we can change all existing expandable enums to have private ctors because of the problems listed above.

Reply:
As discussed offline, we'll not change the behavior of the default constructor but instead add `@Deprecated` annotation to these ctors.